### PR TITLE
74_search_2d_matrix.py

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code

### CODE REVIEW

**Comment by ryujumoto** [comment]
There is no files.

**Comment by Rwatana** [comment]
I changed and check the file exists.

File: 74_search_2d_matrix.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from typing import List
```
Comment (ryujumoto) [inline 74_search_2d_matrix.py:3]: Write time complexity and space complexity.

**Review by ryujumoto — COMMENTED** [review]
Can you solve this with more efficient algorithm?

**Comment by Rwatana** [comment]
I revised it. Please check

File: 74_search_2d_matrix.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# time complexity O(logn)
+# space complexity O(n)
```
Comment (ryujumoto) [inline 74_search_2d_matrix.py]: Space complexity is wrong.

File: 74_search_2d_matrix.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+from typing import List
+
+
+class Solution:
+    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
+        if not matrix:
+            return False
```
Comment (ryujumoto) [inline 74_search_2d_matrix.py:10]: Insert a blank line.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I revised.
Could you tell me the article saying that space complexity. I searched many sites. they show time complexity. But, It doesn't have much information about it.

**Comment by ryujumoto** [comment]
> I revised. Could you tell me the article saying that space complexity. I searched many sites. they show time complexity. But, It doesn't have much information about it.


https://www.geeksforgeeks.org/time-complexity-and-space-complexity/

**Comment by ryujumoto** [comment]
Reference [this code](https://github.com/okuda-seminar/leetcode_answers/blob/main/74_search-a-2d-matrix.py).

**Comment by Rwatana** [comment]
> > I revised. Could you tell me the article saying that space complexity. I searched many sites. they show time complexity. But, It doesn't have much information about it.
> 
> https://www.geeksforgeeks.org/time-complexity-and-space-complexity/

I think I could understand the idea of space complexity.

Is it true that the complexity is the idea that how many units we use when trying the program.

**Comment by ryujumoto** [comment]
Yes, it is.

**Comment by Rwatana** [comment]
I revised please check my code.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**